clone-repos.sh

```diff
@@ -1,16 +1,31 @@
-#!/bin/sh -ex
+#!/bin/sh -eux
 # Clone github repos, and pull to refresh them if they exist
+# vi:shiftwidth=2
 
-! grep "NAME=\"Debian" /etc/os-release > /dev/null
-NOT_DEBIAN=$?
-
-if [ $NOT_DEBIAN = 0 ]; then
-    GCC=gcc-arm-linux-gnueabihf
-else
+if grep -q "NAME=\"Debian" /etc/os-release ; then
     GCC=gcc-4.7-arm-linux-gnueabihf
+else
+    GCC=gcc-arm-linux-gnueabihf
 fi
 
-sudo apt-get -y install rsync git $GCC build-essential qemu kpartx binfmt-support qemu-user-static python bc parted dosfstools
+sudo apt-get -y install rsync git $GCC build-essential kpartx binfmt-support python bc parted dosfstools
+case $DISTROVER in
+  trusty)
+    sudo apt-get -y install qemu qemu-user-static
+    LINUX_URL=https://github.com/talex5
+    LINUX_BRANCH=master
+    XEN_URL=https://github.com/talex5
+    XEN_BRANCH=fix-grant-mapping
+    APPLY_PATCHES=true
+    ;;
+  vivid)
+    sudo apt-get -y install qemu-utils
+    LINUX_URL=https://github.com/infidel
+    LINUX_BRANCH=cubie-vivid
```

Comment:
No, nothing special about it. I can give git://kernel.ubuntu.com/ubuntu/ubuntu-vivid.git a try